saleor/graphql/views.py

```diff
@@ -197,30 +200,33 @@ def handle_query(self, request: HttpRequest) -> JsonResponse:
             # we can calculate the RAW UTF-8 size using the length of
             # response.content of type 'bytes'
             span.set_tag("http.content_length", len(response.content))
-
+            with observability_reporter.api_call_context(request) as api_call:
+                api_call.response = response
+                api_call.report()
             return response
 
     def get_response(
         self, request: HttpRequest, data: dict
     ) -> Tuple[Optional[Dict[str, List[Any]]], int]:
-        execution_result = self.execute_graphql_request(request, data)
-        status_code = 200
-        if execution_result:
-            response = {}
-            if execution_result.errors:
-                response["errors"] = [
-                    self.format_error(e) for e in execution_result.errors
-                ]
-            if execution_result.invalid:
-                status_code = 400
+        with observability_reporter.gql_operation_context() as operation:
+            execution_result = self.execute_graphql_request(request, data)
+            status_code = 200
+            if execution_result:
+                response = {}
+                if execution_result.errors:
+                    response["errors"] = [
+                        self.format_error(e) for e in execution_result.errors
+                    ]
+                if execution_result.invalid:
+                    status_code = 400
```

Comment:
Do you really need a status_code here? Looks like you have  200 when everything is ok, and 400 when something is wrong. Doesn't `response["errors"]` have enough information that something went wrong?
Also, do we really want to use similar http status_code  here? 200/400 could suggest this, but this could be a simple `invalid:true/false` value, right?

Reply:
Agree, fixed